src/ast/Guest.ts

```diff
@@ -37,15 +38,24 @@ export default class Guest extends Node {
       throw new ParserError("expected a guest name", this.lineNumber);
     }
     this.email = token; 
-  } 
+  }
+
   evaluate(context: object[]): void {
-    throw new Error("Method not implemented.");
+    // throw new Error("Method not implemented.");
   }
-  nameCheck(): void {
-    throw new Error("Method not implemented.");
+
+  nameCheck(map: object): void {
+    if (!map.hasOwnProperty(this.id)) {
+      let values = [null, this.name,this.email]; 
+      Object.assign(map, {[this.id ]: values});
```

Comment:
I think this actual value assignment part should happen in `evaluate()`, and this part should just put the key `this.id` into the map

Reply:
@shannonjanehogan  i don't know if you'll be able to see this once I push up changes but evaluate basically just fills in the values in the array. currently I've just initialized the array with null values.